src/Symfony/Bridge/Doctrine/Types/AbstractBinaryUidType.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Types;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\DBAL\Types\GuidType;
+use Symfony\Component\Uid\AbstractUid;
+
+abstract class AbstractBinaryUidType extends GuidType
+{
+    abstract protected function getUidObject(): AbstractUid;
+
+    public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform): string
+    {
+        return $platform->getBinaryTypeDeclarationSQL(
+            [
+                'length' => '16',
+                'fixed' => true,
+            ]
+        );
+    }
+
+    /**
+     * @throws ConversionException
+     */
+    public function convertToPHPValue($value, AbstractPlatform $platform): ?AbstractUid
+    {
+        if (null === $value || '' === $value) {
+            return null;
+        }
+
+        if ($value instanceof AbstractUid) {
+            return $value;
+        }
+
+        try {
+            $uuid = $this->getUidObject()::fromString($value);
+        } catch (\InvalidArgumentException $e) {
+            throw ConversionException::conversionFailed($value, $this->getName());
+        }
+
+        return $uuid;
+    }
+
+    /*
+     * @throws ConversionException
+     */
+    public function convertToDatabaseValue($value, AbstractPlatform $platform): ?string
+    {
+        if (null === $value || '' === $value) {
+            return null;
+        }
+
+        if ($value instanceof AbstractUid) {
+            return $value->toBinary();
+        }
+
+        try {
+            if (\is_string($value) || method_exists($value, '__toString')) {
```

Comment:
in the "else" case, this should return null explicitly, isn't it? (same in the other abstract class)

Reply:
You're right.